### ISSUE TITLE
no-console error when not in browser environment

### ISSUE THREAD

**Issue by zanona** [eslint/eslint#1827]
I currently have no environment set on my eslint configuration file and yet still get `no-console` errors?
Shouldn't those be only triggered when in a browser environment?

```js
/*eslint-env browser*/
console.log(''); // bad
```

```js
//no environment
console.log(''); //good
```

**Comment by nzakas** [comment]
`no-console` is on by default.

**Comment by zanona** [comment]
Yes, but should it?
It should be off by default and `on` only on browser environment.

**Comment by michaelficarra** [comment]
Why would you think that?

**Comment by zanona** [comment]
hey @michaelficarra, because anywhere other than the browser, the use of `console` is quite alright? and commonly used as one good way to print to stdout?
node.js, iojs, phantomjs, etc…

So it makes me wonder why does es-lint establishes the browser environment by default even if there is not environment being set previously?

**Comment by michaelficarra** [comment]
No, most libraries should *not* be emitting anything to a console. In the ones that do (CLIs, log managers), either use `process.stdout.write` or disable the rule. The vast majority of code benefits from having this rule enabled by default.

**Comment by zanona** [comment]
I'm afraid the is no difference between `process.stdout.write` and `console` on [nodejs for example].
Both seem to print to stdoud equally where `console` only offers nice arguments manipulation.
At least that is what is said on the [documentation](http://nodejs.org/api/console.html) ;)

Also, as per es-lint the [documentation](http://eslint.org/docs/rules/no-console.html):

> When Not To Use It (no-console)
> If you're using Node.js, however, console is used to output information to the user and so is not strictly used for debugging purposes. If you are developing for Node.js then you most likely do not want this rule enabled.

Which confirms what I meant.

**Comment by michaelficarra** [comment]
I was not suggesting you actually use `process.stdout.write`, I was just saying it would silence the errors. You should disable the rule in those cases. But my point still stands: nearly all node modules *should not* write to the console. `console.log` calls in those modules are likely left-over printf-style debugging.

**Comment by zanona** [comment]
That makes sense. Even if my module display progress of something on screen it is informative think `npm install` status which is basically debugging information, but also useful to the user.

The only downside I think of at the moment is that there are many types of console output.
`log`, `info`, `error`, `warn` and most recently on webkit browsers, `console.debug` which means, not all that goes through console is necessarily debug information. sometimes you should notify about an error.

In such cases, only `console.log` should be prevented but I am not sure about the others.
Anyway, until this idea is matured with time, I can use a separate module and just name it differently.
After all, on on command-line applications, the only way to communicate with users, is through the console/terminal, so that might not always be a bad thing.

**Comment by nzakas** [comment]
Thanks for the feedback. As @michaelficarra mentioned, in almost every case it makes sense to have `no-console` enabled by default. Node.js is an exception, which is why using the `node` environment disables `no-console`. We're not planning on making any change to the current behavior.
